Client/Frontend/Home/Wallpapers/v2/UI/WallpaperSelectorViewModel.swift

```diff
@@ -0,0 +1,116 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class WallpaperSelectorViewModel {
+
+    enum WallpaperSelectorLayout: Equatable {
+        case compact
+        case regular
+
+        // The maximum number of items to display in the whole section
+        var maxItemsToDisplay: Int {
+            switch self {
+            case .compact:
+                return 6
+            case .regular:
+                return 8
+            }
+        }
+
+        // The maximum number of items to display per row
+        var itemsPerRow: Int {
+            switch self {
+            case .compact:
+                return 3
+            case .regular:
+                return 4
+            }
+        }
+
+        // The maximum number of seasonal items to display
+        var maxNumberOfSeasonalItems: Int {
+            switch self {
+            case .compact:
+                return 3
+            case .regular:
+                return 5
+            }
+        }
+    }
+
+    private var wallpaperManager: WallpaperManagerInterface
+    var openSettingsAction: (() -> Void)
+    var sectionLayout: WallpaperSelectorLayout = .compact // We use the compact layout as default
+    var wallpaperCellModels: [WallpaperCellViewModel] = []
```

Comment:
```suggestion
    var wallpaperCellModels = [WallpaperCellViewModel]()
```
nit: we haven't made a style decision. But, maybe?